text/0000-rust-lang-crates.md

```diff
@@ -0,0 +1,231 @@
+
+### Relation to `std`
+
+In many cases, the above description of the crate lifecycle is complete. But
+some rust-lang crates are destined for std. Usually this will be clear up front.
+
+When a std-destined crate has reached sufficient maturity, the libs subteam can
+call a "final comment period" for moving it into `std` proper. Assuming there
+are no blocking objections, the code is moved into `std`, and the original repo
+is left intact, with the following changes:
+
+- a minor version bump,
+- *conditionally* replacing all definitions with `pub use` from `std` (which
+  will require the ability to `cfg` switch on feature/API availability -- a
+  highly-desired feature on its own).
+
+By re-routing the library to `std` when available we provide seamless
+compatibility between users of the library externally and in `std`. In
+particular, traits and types defined in the crate are compatible across either
+way of importing them.
+
+### Deprecation
+
+At some point a library may become stale -- either because it failed to make it
+out of the nursery, or else because it was supplanted by a superior library. The
+libs subteam can deprecate nursery crates at any time, and can deprecate
+rust-lang crates through an RFC. This is expected to be a rare occurrence.
+
+Deprecated crates move to rust-lang-deprecated and are subsequently minimally
+maintained. Alternatively, if someone volunteers to maintain the crate,
+ownership can be transferred externally.
+
+## Advertising
+
+Part of the reason for having rust-lang crates is to have a clear, short list of
+libraries that are broadly useful, vetted and maintained. But where should this
+list appear?
+
+This RFC doesn't specify the complete details, but proposes a basic direction:
+
+- The crates in rust-lang should appear in the sidebar in the core rustdocs
+  distributed with Rust, along side the standard library. (For nightly releases,
+  we should include the nursery crates as well.)
+
+- The crates should also be published on crates.io, and should somehow be
+*badged*. But the design of a badging/curation system for crates.io is out of
+scope for this RFC.
+
+## Plan for existing crates
+
+There are already a number of non-`std` crates in rust-lang. Below, we give the
+full list along with recommended actions:
+
+### Transfer ownership
+
+Please volunteer if you're interested in taking one of these on!
+
+- rlibc
+- semver
+- threadpool
+
+### Move to rust-lang-nursery
+
+- bitflags
+- getopts
+- glob
+- libc
+- log
+- rand (note, @huonw has a major revamp in the works)
+- regex
+- rustc-serialize (but will likely be replaced by serde or other approach eventually)
+- tempdir (destined for `std` after reworking)
+- uuid
+
+### Move to rust-lang-deprecated
+
+- fourcc: highly niche
+- hexfloat: niche
+- num: this is essentially a dumping ground from 1.0 stabilization; needs a complete re-think.
+- term: API needs total overhaul
+- time: needs total overhaul destined for std
```

Comment:
I would claim that any replacement for this crate would be such a large overhaul that this is basically pre-deprecated. We'll still continue to maintain it and fix bugs here and there, and you won't get a warning message or anything when you use it, it would just move to a new location